### ISSUE TITLE
Pickup and deliver mode - multiple problems

### ISSUE THREAD

**Issue by Martinnygaard** [Stephan-S/FS19_AutoDrive#934]
AutoDrive Version: 1.0.6.9-12
Release or custom zip: Custom
Map (if applicable): All
Vehicle / type (applicable): ALL

**Describe the bug**
All  my trucks that are set to sell goods from my silo will go to the refill point, fill up the trailer and then just wait there. 
It does not matter if i press "continue". It juist waits.
If i stop AD and drive out of the trigger area, it will go on as planned, but then next time it needs to pickup, it waits again.

**If applicable - Steps to reproduce the behavior**
Run any pickup and deliver routes.

**Screenshots**
This truck is full, but has not changed destination to unload point.

![image](https://user-images.githubusercontent.com/26490587/69493564-74830d00-0eb0-11ea-88c5-2d33caaa9c6d.png)

![image](https://user-images.githubusercontent.com/26490587/69493574-98dee980-0eb0-11ea-8072-63769d247bf9.png)

**Comment by Martinnygaard** [comment]
Some more documentation.

As it closes up to the trigger, it stops and fill as soon as possible. (before it accetly reach the trigger):
![image](https://user-images.githubusercontent.com/26490587/69494056-4ead3680-0eb7-11ea-9987-380182f1b882.png)

Then when it is 100% full, it drives on till it acetyl reach the fill trigger - and then just waits.
![image](https://user-images.githubusercontent.com/26490587/69494093-c24f4380-0eb7-11ea-88dd-d6b3080a2a1e.png)

**Comment by Martinnygaard** [comment]
I have two new comments:

Comment 1:
When transporting diesel from _**Fuel - DieselFactory**_ to _**Refinery Diesel Sales**_ it loads the liquid and continue as intented. (The issue above does not exist here?)

It arrives at the destination (_**Refinery Diesel Sales**_) and unloads all the diesel. Then just stops and wait.

When i try to restart AD, i get the following message, that _**Fuel - DieselFactory**_ could not be reached.

![image](https://user-images.githubusercontent.com/26490587/69500682-8ccd4900-0efd-11ea-9099-ad28c11a06cf.png)

When i chagne mode to just the drive mode, it can easily find the destination?
![image](https://user-images.githubusercontent.com/26490587/69500687-9fe01900-0efd-11ea-95aa-e9c43a1fd23c.png)



Comment 2:
If i try to start the route with an Empty trailer, it start heading directly to the unload destination.

![image](https://user-images.githubusercontent.com/26490587/69500805-9d31f380-0efe-11ea-8bec-a42ee2e0c232.png)

**Comment by Martinnygaard** [comment]
Here i am trying to start a route (the trailer is full) And it again say "Diver cant reach....".
If i drive out of the trigger area, it can easily find the destination. Even if i reverca to a point before the trigger, it will manage.

Only when i try to start while it is near the trigger, i can't.


Parked under the trigger - it won't start:
![image](https://user-images.githubusercontent.com/26490587/69501005-e71bd900-0f00-11ea-966b-94c5e26298dc.png)

Parked BEHIND the triggger - It starts:
![image](https://user-images.githubusercontent.com/26490587/69501040-45e15280-0f01-11ea-87c5-2f55cd8679d9.png)

**Comment by Mr-Cake** [comment]
Hi, 

I'm not sure on the first issue, maybe your network doesn't connect to the sell point? There was/is a bug with loading to do with that silo empty setting.

For the last one though, it because AD looks in front of the vehicle for where to start from not where it is now, so if it the point it finds in front of vehicle doesn't connect to destination it will not start.

**Comment by pjr846** [comment]
same thing happening to me. everything was fine till yesterday evening than for some reason they all started acting this way.

**Comment by caplink811** [comment]
let me admit, that the both issues Martinnygaard described above worked in release 1.064 without any problem. (on Saxonia 2.1 with ad courses by Opa Andre I'm able to pickup and deliver diesel as well as adblue, more than 1 time)

**Comment by Martinnygaard** [comment]
> It's related to the use of dual way routes.

I have removed all dual way routes from the drive path. (From load to unload point)

The Diesel delivering truck is still only heading to the offload (sales) point - regardless of it is 0% or 100% filllevel. No matter where on the route i start AD, it goes to unload dest.
Is that dual way related? If so, im not sure i get why.

![AD Waiting 2](https://user-images.githubusercontent.com/26490587/69650923-9a0a4500-106f-11ea-8376-3b4561f50d3f.png)

**Comment by Martinnygaard** [comment]
I noticed this:
fillLevel: 0
leftCapacity: 0
filled: true
empty: true

So somehow or another, AD did not recognize the capacity of my trailer (a modded MKS32)
I found another mod and that works now.

That debugging overlay i really good to figure out what is going on.
Issue is closed.


![image](https://user-images.githubusercontent.com/26490587/69725100-a3ea8180-111d-11ea-84a2-398f570ce833.png)
